packages/transaction-controller/src/utils/history.ts

```diff
@@ -40,16 +56,99 @@ export function updateTransactionHistory(
 
   const currentState = snapshotFromTransactionMeta(transactionMeta);
   const previousState = replayHistory(transactionMeta.history);
-  const historyEntry = generateHistoryEntry(previousState, currentState, note);
+  const newHistoryEntry = generateHistoryEntry(
+    previousState,
+    currentState,
+    note,
+  );
+
+  if (newHistoryEntry.length > 0) {
+    // Casts required here because this list has two separate types of entries:
+    // TransactionMeta and TransactionHistoryEntry. The only TransactionMeta is the first
+    // entry, but TypeScript loses that type information when `slice` is called for some reason.
+    let updatedHistory = transactionMeta.history.slice() as TransactionHistory;
+    updatedHistory.push(newHistoryEntry);
```

Comment:
Minor, could this be achieved with a `...` instead?

Reply:
Yes, it could. I was trying to avoid the need for a type cast here, but it didn't work, so I'll switch it back to `...`.

Done in ee3692d0